core/domain/feedback_services.py

```diff
@@ -357,6 +358,31 @@ def pop_feedback_message_references(user_id, references_length):
         enqueue_feedback_message_email_task(user_id)
 
 
+def update_feedback_message_references(user_id, exploration_id):
+    """Removes feedback message references associated with exploration."""
+    if not user_id:
+        # If user is not logged in this will be True.
+        return
+
+    model = feedback_models.UnsentFeedbackEmailModel.get(user_id, strict=False)
+    if model is None:
+        # Model exists only if user has received feedback on exploration.
+        return
+
+    updated_references = []
+    for reference in model.feedback_message_references:
+        if exploration_id not in reference.values():
+            updated_references.append(reference)
```

Comment:
Actually, we could go more fine-grained here, and fire an event to the backend when a particular message thread is opened by a user. The event contains the last message in that thread. Then we clear just the messages from that message thread, up to and including the last message. Would this make better sense from a user perspective?

Basically, if the user opens one thread, that's not a guarantee that they've opened the other threads with new messages.